scamp/scamp-isr.c

```diff
@@ -264,18 +264,18 @@ INT_HANDLER ap_int()
     sark_lock_free(cpsr, LOCK_MBOX);
 
     if (cmd == SHM_MSG) {
-	vcpu->mbox_mp_cmd = SHM_IDLE;
 
 	sdp_msg_t *msg = sark_msg_get();
 
 	if (msg != NULL) {
+		vcpu->mbox_mp_cmd = SHM_IDLE;
 	    sark_msg_cpy(msg, shm_msg);
 	    msg_queue_insert(msg, 0);
+	    sark_shmsg_free(shm_msg);
 	} else {
 	    sw_error(SW_OPT);
```

Comment:
A possible simple option is to add a vcpu->mbox_mp_cmd = SHM_ERR here, which will then tell the sender that something has changed at least.